### DISCUSSION TITLE
Bring back the CLI from Gitbook legacy

### DISCUSSION THREAD

**Discussion by piranna** [GitbookIO/community#7] [Feature Requests]
### What's your feature/idea?

Gitbook legacy started as a CLI tool, and later Gitbook 2.0 moved to the cloud. The killer feature of Gitbook was that CLI, an open source tool to be able to create books from scratch using markdown, that later could also be published as Github repositories or in the web with Github pages. In fact, I wrote myself [my bachelor thesis](https://github.com/piranna/pfc) using Gitbook, and having my father worked on books publishing himself for 40 years, I contributed with some plugins and PRs helping to make Gitbook a better books self-publishing tool, like justifying text when printing, ensure no orphan or vidow lines exists, moving and printing links at chapters bottom, or proper SVG templating of books covers, between others. Gitbook felt like the last step on projects documentation, and I included Gitbook CLI in all my Node.js projects.

Now, after deprecating Gitbook CLI, all this disappeared. First it got unmaintained while having a huge community behind, and later the repo was fully removed at all (or at least their issues). Only alternative that offer similar features is BookMD, but include a markdown program just to generate documentation of Node.js projects is cumbersome and overkill. Having online publishing is fine, but what we want and need, is local publishing of books on PDF format, online publishing is just a nice-to-have feature, because we already publish our projects documentation and source code in the cloud ourselves by publhising it on Github.

### What's your desired outcome?

Be able to generate PDF books fully offline with a CLI tool, as it was possible with Github legacy before 2.0 version, and being that CLI tool updated and maintained, and being able to extend it with plugins. Gitbook 2.0 should have always been a "Gitbook CLI running on the cloud", not just the only way of using Gitbook and deprecating local usage.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

### Anything else?

Publish the CLI and the core functionality with a proper Open Source license that can allow others to contribute and improve and extend it, and stick to keep maintaining the CLI tool in the future and don't remove/deprecate it again.

**Comment by ChucKN0risK** [comment]
Couldn't agree more. This is definitely something that caught my eye when I began using Gitbook a few weeks ago. Being able to edit a project while being offline and extending native features with custom open source plugin would be a huge pro.

**Comment by ShutdownRepo** [comment]
+1, there are use cases where rebuilding a local instance from the markdown structure would be beneficial, there is no way a gitbook website content would be accessible without Internet connection otherwise. 

In my opinion, it wouldn't even be a risk for the business-model of gitbook as the main advantages of the cloud-based solution is the collaboration/teams/spaces and so forth. Open-sourcing the engine that turns the markdown structure to a website would be an **awesome** move
